modules/refined/src/test/scala/doobie/refined/RefinedSuite.scala

```diff
@@ -80,26 +78,26 @@ class RefinedSuite extends munit.FunSuite {
       _ <- Update0(s"CREATE LOCAL TEMPORARY TABLE TEST (value INT)", None).run
       _ <- sql"INSERT INTO TEST VALUES ($v)".update.run
     } yield ()
-    queryRes.transact(xa).unsafeRunSync()
+    queryRes.transact(xa).assert
   }
 
   test("Query should throw an SecondaryValidationFailed if value does not fit the refinement-type ") {
     secondaryValidationFailedCaught_?(
-      sql"select -1".query[PositiveInt].unique.transact(xa).void.unsafeRunSync()
+      sql"select -1".query[PositiveInt].unique.transact(xa).void.assert
     )
   }
 
   test("Query should return a refined product-type when conversion is possible") {
-    sql"select 1, 1".query[PointInQuadrant1].unique.transact(xa).void.unsafeRunSync()
+    sql"select 1, 1".query[PointInQuadrant1].unique.transact(xa).void.assert
   }
 
   test("Query should throw an SecondaryValidationFailed if object does not fit the refinement-type ") {
     secondaryValidationFailedCaught_?(
-      sql"select -1, 1".query[PointInQuadrant1].unique.transact(xa).void.unsafeRunSync()
+      sql"select -1, 1".query[PointInQuadrant1].unique.transact(xa).void.assert
     )
   }
 
-  private def secondaryValidationFailedCaught_?(query: => Unit) =
-    intercept[SecondaryValidationFailed[?]](query)
+  private def secondaryValidationFailedCaught_?(query: => IO[Unit]) =
+    query.intercept[SecondaryValidationFailed[?]]
```

Comment:
By-name parameter is redundant with effects:
```suggestion
  private def secondaryValidationFailedCaught_?(query: IO[Unit]) =
    query.intercept[SecondaryValidationFailed[?]]
```
To be honest, I'm not sure this function is necessary at all. With the effect assertsions it would probably be clearer to write tests directly like this:
```scala
  test("Query should throw an SecondaryValidationFailed if object does not fit the refinement-type ") {
    sql"select -1, 1".query[PointInQuadrant1].unique.transact(xa)
      .intercept[SecondaryValidationFailed[?]]
  }
```
In my opinion, it looks so much better than all this clutter with `secondaryValidationFailedCaught_?`, doesn't it?

Reply:
Indeed it looks much better without this `secondaryValidationFailedCaught_?`